src/test/java/org/beanpod/switchboard/controller/EncoderControllerTest.java

```diff
@@ -83,60 +95,50 @@ final void testRetrieveAllEncoders() {
   // When a encoder is available in the DB
   @Test
   final void testRetrieveEncoder() {
-    when(encoderDao.findEncoder(EncoderFixture.SERIAL_NUMBER)).thenReturn(Optional.of(encoderDTO));
+    when(encoderDao.findEncoder(user, EncoderFixture.SERIAL_NUMBER))
+        .thenReturn(Optional.of(encoderDTO));
     when(maintainDeviceStatus.maintainStatusField(anyList())).thenReturn(listOfDevices);
 
     ResponseEntity<EncoderDto> actualEncoder =
         encoderController.retrieveEncoder(EncoderFixture.SERIAL_NUMBER);
 
     assertNotNull(actualEncoder);
     assertEquals(200, actualEncoder.getStatusCodeValue());
-    assertEquals(encoder.getSerialNumber(), actualEncoder.getBody().getSerialNumber());
+    assertEquals(
+        encoder.getSerialNumber(),
+        Objects.requireNonNull(actualEncoder.getBody()).getSerialNumber());
   }
 
   // When a encoder is unavailable in the DB
   @Test
   final void testRetrieveEncoderEmpty() {
     assertThrows(
         ExceptionType.DeviceNotFoundException.class,
-        () -> {
-          encoderController.retrieveEncoder("NotAvailable");
-        });
+        () -> encoderController.retrieveEncoder("NotAvailable"));
   }
 
   // When a device is available in the DB
   @Test
   final void testCreateEncoder() {
-    when(deviceService.findDevice(EncoderFixture.SERIAL_NUMBER)).thenReturn(Optional.of(deviceDto));
-    when(encoderDao.save(encoderDTO)).thenReturn(encoderDTO);
-    ResponseEntity response = encoderController.createEncoder(encoderDTO);
+    when(deviceService.findDevice(user, EncoderFixture.SERIAL_NUMBER))
+        .thenReturn(Optional.of(deviceDto));
+    when(encoderDao.save(user, encoderDTO)).thenReturn(encoderDTO);
+    ResponseEntity<EncoderDto> response = encoderController.createEncoder(encoderDTO);
     assertEquals(200, response.getStatusCodeValue());
   }
 
-  @Test
-  final void testCreateEncoderWithoutChannels() {
-    encoderDTO.setOutput(Collections.emptySet());
-    assertThrows(
-        ExceptionType.MissingChannelsException.class,
-        () -> {
-          encoderController.createEncoder(encoderDTO);
-        });
-  }
-
```

Comment:
Why remove this test?

Reply:
I have a feeling @FGRCL might have removed the test by mistake when he merge committed master to the branch. A possible casualty of resolving merge conflicts. I'll add it back.